test/asynchronous/unified_format.py

```diff
@@ -0,0 +1,1573 @@
+                    )
+                )
+            options = parse_collection_or_database_options(spec.get("collectionOptions", {}))
+            self[spec["id"]] = database.get_collection(spec["collectionName"], **options)
+            return
+        elif entity_type == "session":
+            client = self[spec["client"]]
+            if type(client).__name__ != "AsyncMongoClient":
+                self.test.fail(
+                    "Expected entity {} to be of type AsyncMongoClient, got {}".format(
+                        spec["client"], type(client)
+                    )
+                )
+            opts = camel_to_snake_args(spec.get("sessionOptions", {}))
+            if "default_transaction_options" in opts:
+                txn_opts = parse_spec_options(opts["default_transaction_options"])
+                txn_opts = TransactionOptions(**txn_opts)
+                opts = copy.deepcopy(opts)
+                opts["default_transaction_options"] = txn_opts
+            session = client.start_session(**dict(opts))
+            self[spec["id"]] = session
+            self._session_lsids[spec["id"]] = copy.deepcopy(session.session_id)
+            self.test.addAsyncCleanup(session.end_session)
+            return
+        elif entity_type == "bucket":
+            db = self[spec["database"]]
+            kwargs = parse_spec_options(spec.get("bucketOptions", {}).copy())
+            bucket = AsyncGridFSBucket(db, **kwargs)
+
+            # PyMongo does not support AsyncGridFSBucket.drop(), emulate it.
+            @_csot.apply
+            async def drop(self: AsyncGridFSBucket, *args: Any, **kwargs: Any) -> None:
+                await self._files.drop(*args, **kwargs)
+                await self._chunks.drop(*args, **kwargs)
+
+            if not hasattr(bucket, "drop"):
+                bucket.drop = drop.__get__(bucket)
+            self[spec["id"]] = bucket
+            return
+        elif entity_type == "clientEncryption":
+            opts = camel_to_snake_args(spec["clientEncryptionOpts"].copy())
+            if isinstance(opts["key_vault_client"], str):
+                opts["key_vault_client"] = self[opts["key_vault_client"]]
+            # Set TLS options for providers like "kmip:name1".
+            kms_tls_options = {}
+            for provider in opts["kms_providers"]:
+                provider_type = provider.split(":")[0]
+                if provider_type in KMS_TLS_OPTS:
+                    kms_tls_options[provider] = KMS_TLS_OPTS[provider_type]
+            self[spec["id"]] = AsyncClientEncryption(
+                opts["kms_providers"],
+                opts["key_vault_namespace"],
+                opts["key_vault_client"],
+                DEFAULT_CODEC_OPTIONS,
+                opts.get("kms_tls_options", kms_tls_options),
+            )
+            return
+        elif entity_type == "thread":
+            name = spec["id"]
+            thread = SpecRunnerThread(name)
+            thread.start()
```

Comment:
We probably need to add an async version of SpecRunnerThread which creates a task instead of a thread.